### PR TITLE
Github Actions Workflow Refactor

### PR DESCRIPTION
**Pull request by epikichi**
# Description

Introducing Github workflow rework to improve experience.
- Reusable workflows
- Snyk integration

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# This file configures github.com/golangci/golangci-lint.
+
+run:
```
Comment (b00ris) [inline .golangci.yml:3]: Could i ask you to change lint command(in makefile) to use this config?

**Review by b00ris — CHANGES_REQUESTED** [review]
Overall - looks good!

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯

**Review by vcastellm — COMMENTED** [review]
Late LGTM

**Pull request merged**